internal/workflow/processing.go

```diff
@@ -911,18 +916,48 @@ func (w *ProcessingWorkflow) preprocessing(ctx temporalsdk_workflow.Context, tin
 		WorkflowID:        fmt.Sprintf("%s-%s", w.cfg.Preprocessing.Temporal.WorkflowName, uuid.New().String()),
 		ParentClosePolicy: temporalapi_enums.PARENT_CLOSE_POLICY_TERMINATE,
 	})
-	var result preprocessing.WorkflowResult
+	var ppResult preprocessing.WorkflowResult
 	err = temporalsdk_workflow.ExecuteChildWorkflow(
 		preCtx,
 		w.cfg.Preprocessing.Temporal.WorkflowName,
 		preprocessing.WorkflowParams{RelativePath: relPath},
-	).Get(preCtx, &result)
+	).Get(preCtx, &ppResult)
 	if err != nil {
 		return err
 	}
 
-	tinfo.TempPath = filepath.Join(w.cfg.Preprocessing.SharedPath, filepath.Clean(result.RelativePath))
+	tinfo.TempPath = filepath.Join(w.cfg.Preprocessing.SharedPath, filepath.Clean(ppResult.RelativePath))
 	tinfo.IsDir = true
 
+	// Save preprocessing preservation task data.
+	if len(ppResult.PreservationTasks) > 0 {
+		opts := withLocalActivityOpts(ctx)
+		var savePPTasksResult localact.SavePreprocessingTasksActivityResult
+		err = temporalsdk_workflow.ExecuteLocalActivity(
+			opts,
+			localact.SavePreprocessingTasksActivity,
+			localact.SavePreprocessingTasksActivityParams{
+				PkgSvc:               w.pkgsvc,
+				RNG:                  w.rng,
+				PreservationActionID: tinfo.PreservationActionID,
+				Tasks:                ppResult.PreservationTasks,
+			},
+		).Get(opts, &savePPTasksResult)
+		if err != nil {
+			return err
+		}
+
+		switch ppResult.Outcome {
+		case preprocessing.OutcomeSuccess:
+			return nil
+		case preprocessing.OutcomeSystemError:
+			return errors.New("preprocessing workflow: system error")
+		case preprocessing.OutcomeContentError:
+			return errors.New("preprocessing workflow: validation failed")
+		default:
+			return fmt.Errorf("preprocessing workflow: unknown outcome %d", ppResult.Outcome)
+		}
```

Comment:
@djjuhasz, just noticed you moved the outcome check inside the preservation tasks condition too, was that intentional?

Reply:
Yeah, I agree. But I think it makes more sense outside, even if the result doesn't include an outcome, they way you implemented it, it should be success by default, right?

Reply:
Yes, "success" is the default. :)

Reply:
See https://github.com/artefactual-sdps/enduro/pull/966